### PR TITLE
UI/updater: Only run updater on Windows 10+

### PR DESCRIPTION
**Pull request by RytoEX**
### Description

OBS Studio 28 no longer supports Windows 7, 8, or 8.1, and Qt 6 will fail to run on those versions. Users on those systems being offered the update will end up in a state where OBS will not run if they install the update. Let's prevent the updater executable from running on those versions to avoid that scenario.

### Motivation and Context

Want to prevent broken installations.

### How Has This Been Tested?

Compiled locally, though I don't have a test Win7/8 installation to test the updater.

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

- Tweak (non-breaking change to improve existing functionality)

- Breaking change (fix or feature that would cause existing functionality to change)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [ ] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/win-update/updater/updater.cpp [hunk 1]
```diff
@@ -1873,6 +1873,18 @@ int WINAPI wWinMain(HINSTANCE hInstance, HINSTANCE, LPWSTR lpCmdLine, int)
 
 	is32bit = wcsstr(cwd, L"bin\\32bit") != nullptr;
 
+	if (!IsWindows10OrGreater()) {
+		MessageBox(
+			nullptr,
+			L"OBS Studio 28 and newer no longer support Windows 7,"
+			L" Windows 8, or Windows 8.1. You can disable the"
+			L" following setting to opt out of future updates:"
+			L" Settings -> General -> General -> Automatically"
```
Comment (notr1ch) [inline UI/win-update/updater/updater.cpp:1882]: I would recommend using Unicode arrows here to match how OBS itself presents these guides.

Comment (RytoEX) [reply UI/win-update/updater/updater.cpp:1882]: Could you recommend a specific character?

Comment (notr1ch) [reply UI/win-update/updater/updater.cpp:1882]: We use → in the existing texts.

Comment (RytoEX) [reply UI/win-update/updater/updater.cpp]: Pushed changes.

**Pull request merged**